File: javascript/가장큰수.js
function solution(numbers) {
  let ans = numbers
    .map((x) => String(x))
    .sort((x, y) => (x === y ? 0 : parseInt(x + y) > parseInt(y + x) ? -1 : 1))
    .reduce((x, y) => (x += y), '');
  return ans[0] === '0' ? '0' : ans;
}

File: java/가장큰수.java
import java.util.*;
class Solution {
    
    class ComStr implements Comparator<String>{
        public int compare(String a, String b){
            String A = a+b;
            String B = b+a;
            return Integer.parseInt(B)-Integer.parseInt(A);
        }
    }
    
    public String solution(int[] numbers) {
        String answer = "";
        
        int cnt=0;
        for(int i=0;i<numbers.length;i++){
            if(numbers[i]==0){
                cnt++;
            }
        }
        if(cnt==numbers.length){
            return "0";
        }
        
        String []num = new String[numbers.length];
        for(int i=0;i<numbers.length;i++){
            num[i] = String.valueOf(numbers[i]);
        }
        Arrays.sort(num,new ComStr() );
        for(int i=0;i<numbers.length;i++){
            answer+=num[i];
        }
        return answer;
    }
}